sdk/src/types/block/core/wrapper.rs

```diff
@@ -34,9 +36,89 @@ pub struct BlockWrapper {
     latest_finalized_slot: SlotIndex,
     /// The identifier of the account that issued this block.
     issuer_id: IssuerId,
+}
+
+impl BlockHeader {
+    /// The length of the block header.
+    pub const LENGTH: usize = size_of::<Self>();
+
+    pub(crate) fn hash(&self) -> [u8; 32] {
+        let mut bytes = [0u8; Self::LENGTH];
+
+        self.pack(&mut SlicePacker::new(&mut bytes)).unwrap();
+        Blake2b256::digest(bytes).into()
+    }
+}
+
+impl Packable for BlockHeader {
+    type UnpackError = Error;
+    type UnpackVisitor = ProtocolParameters;
+
+    fn pack<P: Packer>(&self, packer: &mut P) -> Result<(), P::Error> {
+        self.protocol_version.pack(packer)?;
+        self.network_id.pack(packer)?;
+        self.issuing_time.pack(packer)?;
+        self.slot_commitment_id.pack(packer)?;
+        self.latest_finalized_slot.pack(packer)?;
+        self.issuer_id.pack(packer)?;
+
+        Ok(())
+    }
+
+    fn unpack<U: Unpacker, const VERIFY: bool>(
+        unpacker: &mut U,
+        protocol_params: &Self::UnpackVisitor,
+    ) -> Result<Self, UnpackError<Self::UnpackError, U::Error>> {
+        let protocol_version = u8::unpack::<_, VERIFY>(unpacker, &()).coerce()?;
+
+        if VERIFY && protocol_version != protocol_params.version() {
+            return Err(UnpackError::Packable(Error::ProtocolVersionMismatch {
+                expected: protocol_params.version(),
+                actual: protocol_version,
+            }));
+        }
+
+        let network_id = u64::unpack::<_, VERIFY>(unpacker, &()).coerce()?;
+
+        if VERIFY && network_id != protocol_params.network_id() {
+            return Err(UnpackError::Packable(Error::NetworkIdMismatch {
+                expected: protocol_params.network_id(),
+                actual: network_id,
+            }));
+        }
+
+        let issuing_time = u64::unpack::<_, VERIFY>(unpacker, &()).coerce()?;
+
+        let slot_commitment_id = SlotCommitmentId::unpack::<_, VERIFY>(unpacker, &()).coerce()?;
+
+        let latest_finalized_slot = SlotIndex::unpack::<_, VERIFY>(unpacker, &()).coerce()?;
+
+        let issuer_id = IssuerId::unpack::<_, VERIFY>(unpacker, &()).coerce()?;
+
+        Ok(Self {
+            protocol_version,
+            network_id,
+            issuing_time,
+            slot_commitment_id,
+            latest_finalized_slot,
+            issuer_id,
+        })
+    }
+}
+
+/// Represent the object that nodes gossip around the network.
+#[derive(Clone, Debug, Eq, PartialEq, Getters, CopyGetters)]
+pub struct BlockWrapper {
+    /// The identifier of the block.
+    #[getset(get_copy = "pub")]
+    id: BlockId,
```

Comment:
Can you put it last please?

Reply:
Just feels right in terms of dependency, you need the other fields to compute it so I think last is more appropriate. Also adding at the bottom of the TIP just feels nicer.

Reply:
k done